File: framework/cpp/src/soda/unittest/bootstrap0.cpp
#include <bits/stdc++.h>
#include "soda/leetcode/leetcode.h"
#include "soda/unittest/unittest.h"

using namespace std;
using namespace soda::leetcode;
using namespace soda::unittest;

// step [1]: implement class Solution
// class Solution {};

namespace {
    const auto __ = []() {
        ios_base::sync_with_stdio(false);
        cin.tie(nullptr);
        return 0;
    }();
}

// step [2]: implement test job
class TestJob : public JobTemplate</*TODO result type*/, /*TODO result serial type*/> {
public:
    ResultType execute(const TestRequest& req, TestResponse& resp) override {
        // TODO
        cerr << "Not implemented" << endl;
    }

    ResultSerialType serialize(const ResultType& res) override {
        return res;
    }

    bool validate(const TestRequest& req, const TestResponse& resp) override {
        return req.getExpected<ResultSerialType>() == resp.getResult<ResultSerialType>();
    }
};

int main()
{
    TestJob job;
    Runner runner;
    runner.run(job);
    return 0;
}

File: framework/scala/src/main/scala/soda/scala/unittest/__Bootstrap0__.scala
package soda.scala.unittest

import java.util._
import java.util.stream._

import soda.leetcode._

import soda.unittest._
import soda.unittest.LoggerHelper.logger

// setp [1]: implement object Solution
// object Solution

// step [2]: implement test job
class __Bootstrap0__ extends JobTemplate[Any, Any] {

    override def execute(req: TestRequest, resp: TestResponse): Any = {
        // TODO
        throw new RuntimeException("Not implemented");
    }

    override def serialize(res: Any): Any = {
        res
    }
    
    override def validate(req: TestRequest, resp: TestResponse): Boolean = {
        req.getExpected(classOf[Object]).equals(resp.getResult(classOf[Object]))
    }

}

object __Bootstrap0__ extends App {
    new ConsoleRunner().run(new __Bootstrap0__())
}
